simulator/simple_dds/dds_service.py

```diff
@@ -0,0 +1,214 @@
+            yield z
+
+    def add_message_handler_methods(self):
+        self.message_handlers['NEW_PARTICIPANT'] = self.append_remote_participant
+        self.message_handlers['NEW_TOPIC'] = self.append_remote_topic
+        self.message_handlers['NEW_DATA'] = self.append_data_object
+        self.message_handlers['SEND_ALL_DATA'] = self.send_full_domain_data
+        self.message_handlers['ALL_DATA'] = self.receive_full_domain_data
+
+    def append_remote_participant(self, r_participant):
+        handle = r_participant.get_instance_handle()
+        if handle not in self.participants and handle not in self.handles:
+            self.handles[handle] = r_participant
+            self.participants[handle] = r_participant
+
+    def append_remote_topic(self, r_topic):
+        topic_name = r_topic.get_name()
+        if not self.topic_exists(topic_name):
+            handle = r_topic.get_instance_handle()
+            self.handles[handle] = r_topic
+            self.topics[topic_name] = r_topic
+        else:
+            self.resolve_topic_conflict(r_topic)
+
+    def resolve_topic_conflict(self, topic):
+        # TODO: Completar este método.
+        # O tópico com a instance handle menor tem prioridade.
+        # Caso o serviço local tenha prioridade, é necessário informar os outros nodos.
+        pass
+
+    def append_data_object(self, new_data):
+        handle = new_data.get_instance_handle()
+        self.data_objects[handle] = new_data
+        if handle not in self.handles:
+            self.handles[handle] = new_data
+        self.send_data_object_to_all_participants(new_data)
+        self.attach_data_object_to_topic(new_data)
+
+    def send_data_object_to_all_participants(self, data_object):
+        for participant in self.local_participants.values():
+            participant.update_all_subscribers(data_object)
+
+    def attach_data_object_to_topic(self, data_object):
+        topic_name = data_object.get_topic_name()
+        if self.topic_exists(topic_name):
+            self.topics[topic_name].attach_data_object(data_object)
+
+    def send_full_domain_data(self, to_address):
+        local_data = []
+        for key, value in self.participants.items():
+            packet = ('NEW_PARTICIPANT', value)
+            local_data.append(packet)
+        for key, value in self.topics.items():
+            packet = ('NEW_TOPIC', value)
+            local_data.append(packet)
+        for key, value in self.data_objects.items():
+            packet = ('NEW_DATA', value)
+            local_data.append(packet)
+        msg = ('ALL_DATA', local_data)
+        self.driver.async_function_call(['send', to_address, msg])
+    
+    def receive_full_domain_data(self, r_data):
+        # TODO: Remove this hack.
+        # HACK ALERT I JUST WANT THIS TO WORK
+        for element in r_data:
+            envelope = [0,0, element]
+            for z in self.interpret_data(envelope):
+                z
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Statement seems to have no effect](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5588062)

Reply:
Corrected.